File: python/141-150/Evaluate Reverse Polish Notation.py
    def evalRPN(self, tokens):
        stack=[]
        operator = ['+','-','*','/']
        for i in tokens:
            if i in operator:
                op2 = stack.pop()
                op1 = stack.pop()
                if i=='+':
                    stack.append(op1+op2)
                elif i=='-':
                    stack.append(op1-op2)
                elif i=='*':
                    stack.append(op1*op2)
                elif i=='/':
                    stack.append(op1/op2)  #have some error in /
            else:
                stack.append(int(i))
        return stack[0]

File: Cpp/141-150/Evaluate Reverse Polish Notation.cpp
/**
 * runtime: 16ms
 * error: 1
 * 
 */ 

class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> myStack;
        if(tokens.size()==0) return 0;
        for(int i=0; i<tokens.size(); i++){
            if(tokens[i]=="+"||tokens[i]=="-"||tokens[i]=="*"||tokens[i]=="/"){
                int a = myStack.top();
                myStack.pop();
                int b=  myStack.top();
                myStack.pop();
                int temp;
                if(tokens[i]=="+")
                    temp = a+b;
                else if(tokens[i]=="*")
                    temp = a*b;
                else  if(tokens[i]=="-")
                    temp = b-a;
                else temp = b/a;
                myStack.push(temp);
            }
            else myStack.push(atoi(tokens[i].c_str()));
        }
        return myStack.top();
    }
};
